### PR TITLE
1534 type hints numpy 1 20

### LINKED ISSUE #1534: static type checks fail since numpy 1.20
the recent numpy update (https://github.com/numpy/numpy/releases/tag/v1.20.0) breaks the pull request's type checks https://github.com/Project-MONAI/MONAI/pull/1532/checks?check_run_id=1807350179#step:7:284

### PR DESCRIPTION
**Pull request by wyli**
Fixes #1534

### Description
non-breaking changes to fix the type check issues

mainly relaxing the type constraints, referring to 
- https://github.com/numpy/numpy/blob/ac7676930a8ea25502c2856fa364c0cef0913ae3/numpy/__init__.pyi
- https://github.com/numpy/numpy/tree/ac7676930a8ea25502c2856fa364c0cef0913ae3/numpy/typing

### Status
**Ready**

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [x] Integration tests passed locally by running `./runtests.sh --codeformat --coverage`.
- [x] Quick tests passed locally by running `./runtests.sh --quick`.
- [x] In-line docstrings updated.

### CODE REVIEW

**Review by Nic-Ma — APPROVED** [review]
Hi @wyli ,

Thanks for your quick fix for all the new type hints issues.
Looks good to me except for several minor comments.

Thanks.

File: monai/transforms/croppad/array.py [hunk 1]
```diff
@@ -219,10 +220,10 @@ class SpatialCrop(Transform):
 
     def __init__(
         self,
-        roi_center: Optional[Sequence[int]] = None,
-        roi_size: Optional[Sequence[int]] = None,
-        roi_start: Optional[Sequence[int]] = None,
-        roi_end: Optional[Sequence[int]] = None,
+        roi_center: Union[Sequence[int], np.ndarray, None] = None,
```
[the remark this answers is not in the document]
Comment (wyli) [reply monai/transforms/croppad/array.py:223]: they are equivalent https://docs.python.org/3/library/typing.html#typing.Optional

[the remark this answers is not in the document]
Comment (Nic-Ma) [reply monai/transforms/croppad/array.py:223]: Yes, both are OK for me, just almost all the other places in MONAI use `Optional` for `None` type hints.
Thanks.

**Pull request merged**